tests/kubernetes-e2e.rs

```diff
@@ -0,0 +1,725 @@
+        )
+        .await?;
+
+    let control_test_pod = framework
+        .test_pod(test_pod::Config::from_pod(&make_test_pod(
+            "test-vector-test-pod",
+            "test-pod-control",
+            "echo CONTROL_MARKER",
+            vec![],
+        ))?)
+        .await?;
+    framework
+        .wait(
+            "test-vector-test-pod",
+            vec!["pods/test-pod-control"],
+            WaitFor::Condition("initialized"),
+            vec!["--timeout=30s"],
+        )
+        .await?;
+
+    let mut log_reader = framework.logs("test-vector", "daemonset/vector")?;
+    smoke_check_first_line(&mut log_reader).await;
+
+    // Read the log lines until the reasoable amount of time passes for us
+    // to be confident that vector shoud've picked up the excluded message
+    // if it wasn't fitlering it.
+    let mut got_control_marker = false;
+    let mut lines_till_we_give_up: usize = 10000;
+    let (stop_tx, mut stop_rx) = futures::channel::mpsc::channel(0);
+    loop {
+        let line = tokio::select! {
+            result = stop_rx.next() => {
+                result.unwrap();
+                log_reader.kill()?;
+                continue;
+            }
+            line = log_reader.read_line() => line,
+        };
+        let line = match line {
+            Some(line) => line,
+            None => break,
+        };
+        println!("Got line: {:?}", line);
+
+        lines_till_we_give_up -= 1;
+        if lines_till_we_give_up <= 0 {
+            println!("Giving up");
+            log_reader.kill()?;
+            break;
+        }
+
+        if !line.starts_with("{") {
+            // This isn't a json, must be an entry from Vector's own log stream.
+            continue;
+        }
+
+        let val = parse_json(&line)?;
+
+        if val["kubernetes"]["pod_namespace"] != "test-vector-test-pod" {
+            // A log from something other than our test pod, pretend we don't
+            // see it.
+            continue;
+        }
+
+        // Ensure we got the log event from the control pod.
+        assert_eq!(val["kubernetes"]["pod_name"], "test-pod-control");
+
+        // Ensure the test sanity by validating that we got the control marker.
+        // If we get an excluded marker here - it's an error.
+        assert_eq!(val["message"], "CONTROL_MARKER");
+
+        if got_control_marker {
+            // We've already seen one control marker! This is not good, we only
+            // emitted one.
+            panic!("control marker seen more than once");
+        }
+
+        // Remember that we've seen a control marker.
+        got_control_marker = true;
+
+        // Request termination in a while.
+        let mut stop_tx = stop_tx.clone();
+        tokio::spawn(async move {
+            // Wait for two minutes - a reasonable time for vector internals to
+            // pick up new `Pod` and collect events from them in idle load.
+            let duration = std::time::Duration::from_secs(120);
```

Comment:
The same with before mentioned `Succeeded` pod phase. We could decrease some of that delay.

Reply:
This one is racy by nature - the longer we wait, the more certainty we have that the excluded `Pod` logs weren't picked up because it was excluded, rather than them being just slow to load.
We can think about other ways to assert that, but this one seemed very straightforward and fair.

Reply:
> This one is racy by nature 

That's true. The wait for `Succeeded` was to be used to achieve same/greater level of probability. Unfortunately it's not possible as you commented before.